src/rapidhtml/app.py

```diff
@@ -1,13 +1,40 @@
 from __future__ import annotations
 
 import typing
-
+import inspect
 import uvicorn
 
 from starlette.applications import Starlette
+from pathlib import Path
 
 from rapidhtml.tags import Script
-from rapidhtml.routing import RapidHTMLRouter
+from rapidhtml.routing import RapidHTMLRouter, RapidHTMLWSEndpoint
+
+JS_RELOAD_SCRIPT = """           
+let active = false;
+sock = new WebSocket(`ws://${window.location.host}/live-reload`);
+sock.onopen = function (event) {
+    console.log(`connected`);
+    active = true;
+};
+
+sock.onclose = function (event) {
+    console.log(`disconnected... reloading.`);
+    if (active) {
+        setTimeout(function () {
```

Comment:
Is `setTimeout` necessary? I can remove it with seemingly no ill effects. If it is necessary, we need to reduce the timeout to something much smaller (I was able to set it to 0 with no issues). Having to wait five seconds for the page to reload isn't ideal